common/src/main/java/com/wynntils/models/players/PlayerModel.java

```diff
@@ -192,8 +194,13 @@ private void clearUserCache() {
     }
 
     private void clearTextureCache() {
-        cosmeticTextures.forEach((uuid, array) -> Arrays.stream(array)
-                .forEach(l -> McUtils.mc().getTextureManager().release(l)));
+        for (ResourceLocation[] locations : cosmeticTextures.values()) {
```

Comment:
Just checking here, do the outermost for loop as need to happen when called? Or phrased differently, could not the entire clearTextureCache() be executed on the main loop?

(Maybe I'm missing some race condition on cosmeticTextures here)

Reply:
Wouldn't this create lag spikes in the first place? Why do we need to run on next tick?

Reply:
if it doesn't run on the client thread, it has a chance of causing texture corruption issues (as far as i know)

this PR is meant to fix the reports of bugged out cape textures that we've been receiving